setupext.py

```diff
@@ -101,6 +100,31 @@ def download_or_cache(url, sha):
     return BytesIO(data)
 
 
+def get_and_extract_tarball(urls, sha, dirname):
```

Comment:
For clarity:
```suggestion
def download_and_extract_tarball(urls, sha, dirname):
```

Reply:
It doesn't necessarily download it, it should be `download_or_cache_and_extract_tarball` (cf. `download_or_[get_from_]cache`) but that's quite a mouthful...

Reply:
Then let's take that into the docstring. It's missing a description anyway :smirk:. See below.